crates/flux-middle/src/pretty.rs

```diff
@@ -378,10 +383,14 @@ impl PrettyCx<'_> {
     pub fn hide_sorts(self, b: bool) -> Self {
         Self { hide_sorts: b, ..self }
     }
+
+    pub fn with_genv(self, genv: GlobalEnv<'genv, 'tcx>) -> Self {
+        Self { genv: Some(genv), ..self }
+    }
```

Comment:
nit, it's a bit annoying you have to construct it `PrettyCx::default(genv.tcx()).with_genv(genv)`, so maybe we can make this a constructor

```suggestion
    pub fn default_with_genv(genv: GlobalEnv<'genv, 'tcx>) -> Self {
        Self { genv: Some(genv), Self::default(genv.tcx()) }
    }
```

or 

```rust
    pub fn with_genv(genv: GlobalEnv<'genv, 'tcx>) -> Self {
        Self { genv: Some(genv), Self::default(genv.tcx()) }
    }
```